flink-runtime/src/main/java/org/apache/flink/streaming/api/datastream/ConnectedStreams.java

```diff
@@ -69,12 +71,14 @@ public class ConnectedStreams<IN1, IN2> {
     protected final StreamExecutionEnvironment environment;
     protected final DataStream<IN1> inputStream1;
     protected final DataStream<IN2> inputStream2;
+    protected boolean isEnableAsyncState;
 
     protected ConnectedStreams(
             StreamExecutionEnvironment env, DataStream<IN1> input1, DataStream<IN2> input2) {
         this.environment = requireNonNull(env);
         this.inputStream1 = requireNonNull(input1);
         this.inputStream2 = requireNonNull(input2);
+        this.isEnableAsyncState = false;
```

Comment:
The `isEnableAsyncState` should be initialized considering `input1` and `input2`